File: app/src/main/java/com/mxt/anitrend/view/activity/detail/FavouriteActivity.java
package com.mxt.anitrend.view.activity.detail;

import android.os.Bundle;

import androidx.annotation.Nullable;
import com.mxt.anitrend.adapter.pager.detail.FavouritePageAdapter;
import com.mxt.anitrend.base.custom.activity.ActivityBase;
import com.mxt.anitrend.databinding.ActivityPagerGenericBinding;
import com.mxt.anitrend.model.entity.anilist.Favourite;
import com.mxt.anitrend.presenter.base.BasePresenter;

/**
 * Created by max on 2017/12/14.
 */

public class FavouriteActivity extends ActivityBase<Favourite, BasePresenter> {

    private ActivityPagerGenericBinding binding;

    private FavouritePageAdapter pageAdapter;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityPagerGenericBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        mSearchView = binding.customToolbar.searchView;
        setSupportActionBar(binding.customToolbar.toolbar);
        setPresenter(new BasePresenter(this));
        setViewModel(true);
    }

    @Override
    protected void onPostCreate(@Nullable Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);
        onActivityReady();
    }

    /**
     * Make decisions, check for permissions or fire background threads from this method
     * N.B. Must be called after onPostCreate
     */
    @Override
    protected void onActivityReady() {
        pageAdapter = new FavouritePageAdapter(getSupportFragmentManager(), getApplicationContext());
        pageAdapter.setParams(getIntent().getExtras());
        updateUI();
    }

    @Override
    protected void updateUI() {
        binding.contentMain.pageContainer.setAdapter(pageAdapter);
        binding.contentMain.pageContainer.setOffscreenPageLimit(offScreenLimit);
        binding.customTab.smartTab.setViewPager(binding.contentMain.pageContainer);
    }

    @Override
    protected void makeRequest() {

    }
}

File: app/src/main/java/com/mxt/anitrend/view/activity/detail/FavouriteActivity.kt
package com.mxt.anitrend.view.activity.detail

import android.os.Bundle
import android.view.MenuItem
import com.google.android.material.tabs.TabLayoutMediator
import com.mxt.anitrend.R
import com.mxt.anitrend.adapter.pager.detail.FavouritePageAdapter
import com.mxt.anitrend.databinding.ActivityPagerGenericBinding
import com.mxt.anitrend.view.activity.CommonActivity

class FavouriteActivity : CommonActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val binding = ActivityPagerGenericBinding.inflate(layoutInflater)
        setContentView(binding.root)
        setSupportActionBar(binding.customToolbar.toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val pageAdapter =
            FavouritePageAdapter(this, applicationContext).apply {
                params = intent.extras ?: Bundle.EMPTY
            }
        binding.contentMain.pageContainer.adapter = pageAdapter
        binding.contentMain.pageContainer.offscreenPageLimit = 3
        TabLayoutMediator(binding.customTab.smartTab, binding.contentMain.pageContainer) { tab, position ->
            tab.text = pageAdapter.getPageTitle(position)
        }.attach()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressedDispatcher.onBackPressed()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
